propagation/propagation.go

```diff
@@ -1,3 +1,7 @@
+// Package propagation includes types and functions for marshalling and unmarshalling trace
+// context headers between various supported formats and an internal representation. It
+// provides support for traces that cross process boundaries with support for interoperability
+// between various kinds of trace context header formats.
 package propagation
```

Comment:
Do you think it would be worth while to add an `IsValid()` and `IsEmpty()` function to the `PropagationContext` type? Looking at the various possible return values from the unmarshaling functions, it might be kind to provide easy and obvious ways of doing extra validation on the context. I am especially eager to consider this since an "empty" context returned by the w3c unmarshal code will have `00000000000000000000000000000000` as the `TraceID`, which will prohibit you from using an equality comparison with an empty struct to test for the zero value.

Reply:
I like `IsValid()`. I'm not sure how `IsEmpty()` would differ. For now, I've added `IsValid()` which just checks for the existence of a parent id and a trace id that are not the zero value.